modules/generate-reference-doc/src/main/scala/scala/cli/doc/ReferenceDocUtils.scala

```diff
@@ -0,0 +1,46 @@
+package scala.cli.doc
+
+import caseapp.HelpMessage
+
+import java.util.stream.IntStream
+
+import scala.build.internal.util.ConsoleUtils.*
+import scala.jdk.StreamConverters.*
+
+object ReferenceDocUtils {
+  extension (s: String) {
+    def consoleToFence: String =
+      s
+        .lines()
+        .toScala(scala.List)
+        .fold("") { (acc, line) =>
+          val maybeOpenFence =
+            if line.contains(Console.BOLD) then
+              """```sh
+                |""".stripMargin
+            else if line.contains(ScalaCliConsole.GRAY) then
+              """```scala
+                |""".stripMargin
+            else ""
+          val maybeCloseFence =
+            if line.contains(Console.RESET) then
+              """
+                |```""".stripMargin
+            else ""
+          val newLine = s"$maybeOpenFence${line.noConsoleKeys}$maybeCloseFence"
+          if acc.isEmpty then newLine
+          else s"""$acc
+                  |$newLine""".stripMargin
+        }
+  }
+  extension (helpMessage: HelpMessage) {
+    def referenceDocMessage: String = helpMessage.message.consoleToFence.noConsoleKeys
+    def referenceDocDetailedMessage: String = {
+      val msg =
+        if helpMessage.detailedMessage.nonEmpty then helpMessage.detailedMessage
+        else helpMessage.message
+      msg.consoleToFence.noConsoleKeys
```

Comment:
```suggestion
      msg.consoleToFence
```
is this needed? Console keys are dropped in `consoleToFence`

Reply:
indeed it's not needed.
should be fixed.